File: articles/counterfactual-reasoning-in-algorithmic-systems/fortran/counterfactual_threshold.f90
program counterfactual_threshold
  implicit none
  real :: original, counterfactual, threshold
  original = 0.57
  counterfactual = 0.65
  threshold = 0.62
  print *, 'original_favorable=', original >= threshold
  print *, 'counterfactual_favorable=', counterfactual >= threshold
  print *, 'decision_flipped=', (original < threshold) .and. (counterfactual >= threshold)
end program counterfactual_threshold

File: articles/counterfactual-reasoning-in-algorithmic-systems/typescript/counterfactual_threshold.ts
type Label = "favorable" | "not_favorable";

function label(score: number, threshold: number): Label {
  return score >= threshold ? "favorable" : "not_favorable";
}

const original = label(0.57, 0.62);
const counterfactual = label(0.65, 0.62);
console.log({ original, counterfactual, decisionFlipped: original !== counterfactual });
